project/database/security.py

```
@@ -0,0 +1,129 @@
+
+    def encrypt_text(self, text: str) -> str:
+        """Encrypt sensitive text data with authenticated encryption."""
+        if not text:
+            return ""
+        try:
+            # Fernet provides authenticated encryption (AES 128 + HMAC)
+            encrypted_bytes = self._cipher.encrypt(text.encode("utf-8"))
+            return base64.urlsafe_b64encode(encrypted_bytes).decode("ascii")
+        except Exception as e:
+            logger.error(f"Encryption failed: {e}")
+            raise ValueError("Failed to encrypt data")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)